tests/test_hive.py

```diff
@@ -0,0 +1,190 @@
+        self.kerberos_env = kerberos_env
+        self.hive_agent_ip = hive_agent_ip
+        self.hive_agent_hostname = hive_agent_hostname
+
+
+class HadoopSetup:
+    def __init__(self, kerberos_setup, hdfs_base_url, hive_config_url):
+        self.kerberos_setup = kerberos_setup
+        self.hdfs_base_url = hdfs_base_url
+        self.hive_config_url = hive_config_url
+
+
+def _get_agent_ip():
+    
+    def _is_private_agent(node):
+        return node["type"] == "agent" and not node["attributes"]
+
+    nodes = sdk_cmd.get_json_output("node --json")
+    
+    for node in nodes:
+        if _is_private_agent(node):
+            agent_ip = node["hostname"]
+            break
+    return agent_ip
+
+
+@pytest.fixture(scope='module')
+def kerberos_setup():
+    try:
+        hive_agent_ip = _get_agent_ip()
+
+        ok, stdout = sdk_cmd.agent_ssh(hive_agent_ip, "echo $HOSTNAME")
+        if not ok:
+            raise Exception("Failed to get agent hostname, stdout: {}".format(stdout))
+        log.info("got hostname: '{}'".format(stdout))
+        hive_agent_hostname = stdout.rstrip()
+
+        primaries = ["alice", "hdfs", "HTTP", "yarn", "hive", "sentry"]
+        principals = [ "{primary}/{instance}@{REALM}".format(
+            primary=primary,
+            instance=hive_agent_hostname,
+            REALM=sdk_auth.REALM
+        ) for primary in primaries ]
+        principals.append(GENERIC_HDFS_USER_PRINCIPAL)
+        principals.append(ALICE_PRINCIPAL)
+
+        kerberos_env = sdk_auth.KerberosEnvironment()
+        kerberos_env.add_principals(principals)
+        kerberos_env.finalize()
+        yield KerberosSetup(kerberos_env, hive_agent_ip, hive_agent_hostname)
+
+    finally:
+        kerberos_env.cleanup()
+
+
+def _hdfs_cmd(cmd):
+    rc, _, _ = sdk_cmd.marathon_task_exec(HIVE_APP_ID, "/usr/local/hadoop/bin/hdfs dfs -{}".format(cmd))
+    if rc != 0:
+        raise Exception("HDFS command failed with code {}: {}".format(rc, cmd))
+
+
+def _upload_hadoop_config(file_name, hive_agent_hostname):
+    template_local_path = os.path.join(THIS_DIR, 'resources', "{}.template".format(file_name))
+    with open(template_local_path, 'r') as f:
+        template_contents = f.read()
+
+    with tempfile.TemporaryDirectory() as temp_dir:
+        local_path = os.path.join(temp_dir, file_name)
+        with open(local_path, 'w') as fp:
+            fp.write(template_contents.replace("{{HOSTNAME}}", hive_agent_hostname))
+        url = spark_utils.upload_file(local_path)
+    return url
+
+
+@pytest.fixture(scope='module')
+def hadoop_setup(kerberos_setup):
+    try:
+        # TODO: replace with Marathon readiness check in kdc app?
+        #log.info("Waiting for kdc.marathon.mesos to resolve ...")
+        #time.sleep(10)
+
+        # Run the Hive Docker image in Marathon
+        curr_dir = os.path.dirname(os.path.realpath(__file__))
+        app_def_path = "{}/resources/hadoop-hive-kerberos.json".format(curr_dir)
```

Comment:
For many of our other marathon applications, we specify the app config directly in the python code. Is there a motivation for doing it like this?